lib/ecmascript.ts

```diff
@@ -3591,65 +3609,74 @@ export function UnbalanceDurationRelative(
       break;
     case 'month':
       {
-        if (!calendar) throw new RangeError('a starting point is required for months balancing');
+        // These conditions are effectively identical (as relativeTo being defined forces calendar to be defined)
+        // but checking for their truthiness here helps TypeScript and avoids the need for casts below.
+        if (!relativeTo || !calendar) throw new RangeError('a starting point is required for months balancing');
         // balance years down to months
         const dateAdd = calendar.dateAdd;
         const dateUntil = calendar.dateUntil;
-        let relativeToDateOnly: Temporal.PlainDateLike = relativeTo as Temporal.PlainDateLike;
         while (MathAbs(years) > 0) {
-          const newRelativeTo = CalendarDateAdd(calendar, relativeToDateOnly, oneYear, undefined, dateAdd);
+          const newRelativeTo = CalendarDateAdd(calendar, relativeTo, oneYear, undefined, dateAdd);
           const untilOptions = ObjectCreate(null);
           untilOptions.largestUnit = 'month';
-          const untilResult = CalendarDateUntil(calendar, relativeToDateOnly, newRelativeTo, untilOptions, dateUntil);
+          const untilResult = CalendarDateUntil(calendar, relativeTo, newRelativeTo, untilOptions, dateUntil);
           const oneYearMonths = GetSlot(untilResult, MONTHS);
-          relativeToDateOnly = newRelativeTo;
+          relativeTo = newRelativeTo;
           months += oneYearMonths;
           years -= sign;
         }
       }
       break;
     case 'week':
-      if (!calendar) throw new RangeError('a starting point is required for weeks balancing');
+      // These conditions are effectively identical (as relativeTo being defined forces calendar to be defined)
+      // but checking for their truthiness here helps TypeScript and avoids the need for casts below.
+      if (!relativeTo || !calendar) throw new RangeError('a starting point is required for weeks balancing');
```

Comment:
Should we use an assertion function here as an alternative to changing the `if` condition?

Reply:
Sounds good.

These cases actually can benefit from a stricter assertion (assertExists) for which we can also include some optional runtime type-checks (that get compiled out in prod builds).